app/services/webflow_sync/api.rb

```diff
@@ -32,21 +53,31 @@ def create_item(record, collection_slug) # rubocop:disable Metrics/MethodLength
 
     def update_item(record, collection_slug)
       collection = find_webflow_collection(collection_slug)
-      response = client.update_item(
-        { '_cid' => collection['_id'], '_id' => record.webflow_item_id },
-        record.as_webflow_json.reverse_merge(_archived: false, _draft: false), live: true
-      )
+      response = make_request(:update_item, { '_cid' => collection['_id'], '_id' => record.webflow_item_id },
+                              record.as_webflow_json.reverse_merge(_archived: false, _draft: false), live: true)
+
       puts "Updated #{record.inspect} in #{collection_slug}"
       response
     end
 
     def delete_item(collection_slug, webflow_item_id)
       collection = find_webflow_collection(collection_slug)
-      response = client.delete_item({ '_cid' => collection['_id'], '_id' => webflow_item_id })
+      response = make_request(:delete_item, { '_cid' => collection['_id'], '_id' => webflow_item_id })
+      # When we remove an item from WebFlow, its WebFlow page does not immediately get removed (webflowsite.com/item-slug).
+      # If we trigger publish for all domains, the page gets immediately removed.
```

Comment:
When the item is removed from WebFlow, it's still visible throughout the WebFlow site (probably due to some caching). To remove the item immediately from the WebFlow site, we need to publish the site.